core/src/main/java/com/linecorp/armeria/common/util/BlockingTaskExecutorBuilder.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.Flags;
+
+/**
+ * Provides a builder that is useful for creating a {@link ScheduledExecutorService}.
+ *
+ * @see CommonPools#blockingTaskExecutor()
+ */
+public final class BlockingTaskExecutorBuilder {
+
+    private String threadNamePrefix = "armeria-blocking-tasks";
+    private int numThreads = Flags.numCommonBlockingTaskThreads();
+    private long keepAliveTime = 60;
+    private boolean allowCoreThreadTimeout = true;
+    private boolean daemon = true;
+    private int priority = Thread.NORM_PRIORITY;
+    private Function<? super Runnable, ? extends Runnable> taskFunction = Function.identity();
+
+    BlockingTaskExecutorBuilder() {}
+
+    /**
+     * Sets the prefix of thread names.
+     */
+    public BlockingTaskExecutorBuilder threadNamePrefix(String threadNamePrefix) {
+        requireNonNull(threadNamePrefix, "threadNamePrefix");
+        this.threadNamePrefix = threadNamePrefix;
+        return this;
+    }
+
+    /**
+     * Sets the number of blocking task executor threads.
+     */
+    public BlockingTaskExecutorBuilder numThreads(int numThreads) {
+        checkArgument(numThreads >= 0, "numThreads: %s (expected: >= 0)", numThreads);
+        this.numThreads = numThreads;
+        return this;
+    }
+
+    /**
+     * Sets the flag to permit thread timeout.
+     */
+    public BlockingTaskExecutorBuilder allowCoreThreadTimeout(boolean allowCoreThreadTimeout) {
+        this.allowCoreThreadTimeout = allowCoreThreadTimeout;
+        return this;
+    }
```

Comment:
In `ScheduledThreadPoolService`, every thread is a core thread, so it's probably fine to remove `Core`.